microovn/api/types/ovsdb.go

```diff
@@ -0,0 +1,128 @@
+package types
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/canonical/lxd/shared"
+	"github.com/canonical/microcluster/client"
+)
+
+type OvsdbSchemaFetchError int
+
+const (
+	OvsdbSchemaFetchErrorNone OvsdbSchemaFetchError = iota
+	OvsdbSchemaFetchErrorGeneric
+	OvsdbSchemaFetchErrorNotSupported
+)
+
+// OvsdbSchemaNodeStatus contains information about OVSDB schema status version on an individual host
+type OvsdbSchemaNodeStatus struct {
+	Host            string                `json:"host"`            // Hostname or network address of the cluster member
+	ExpectedVersion string                `json:"expectedVersion"` // OVSDB schema version that the Host expects to be running
+	Error           OvsdbSchemaFetchError `json:"error"`           // Indication whether there was an error fetch ExpectedVersion from this Host
+}
+
+// OvsdbSchemaClusterStatus groups data about the OVSDB cluster schema from multiple cluster members
+type OvsdbSchemaClusterStatus struct {
+	Nodes          []OvsdbSchemaNodeStatus `json:"nodes"`          // List of cluster nodes and their individual statuses
+	RunningVersion string                  `json:"runningVersion"` // currently running schema version
+}
+
+// UpgradeRequired method returns "true" if there's any discrepancy between currently running OVSDB schema version
+// and a schema version expected by the individual hosts in the cluster. It also returns "true", if any of the cluster
+// members had Error in their OvsdbSchemaNodeStatus as it may indicate either a host that needs an attention because
+// it's down, or it's an older version of the MicroOVN that does not support API for fetching expected schema version.
+//
+// If all values match and there was no error, this method returns "false".
+//
+// This method returns error if Nodes is an empty list.
+func (s *OvsdbSchemaClusterStatus) UpgradeRequired(ctx context.Context) (bool, error) {
```

Comment:
This is a method implemented directly on the API result type `OvsdbSchemaClusterStatus`. I think it makes more sense to implement is as method, rather than a free function somewhere else that takes `OvsdbSchemaClusterStatus` as an argument. I'm open to suggestions on alternative implementation, thought it may not matter that much if we end up re-implementing `getSchemaVersionStatus` API, this type may go away entirely.